ext/TrixiNLsolveExt.jl

```diff
@@ -95,11 +95,11 @@ function Trixi.solve_a_butcher_coeffs_unknown!(a_unknown, num_stages, monomial_c
         # Due to the nature of the nonlinear solver, different initial guesses can lead to 
         # small numerical differences in the solution.
 
-        x0 = 0.1 .* rand(rng, num_stages - 2)
+        x0 = 0.1f0 .* rand(rng, num_stages - 2)
```

Comment:
This does not help. `rand` generates `Float64` and `0.1f0` cannot be represented exactly. To make this general, you should get the floating point type `RealT` from somewhere and then use something like

```suggestion
        x0 = convert(RealT, 0.1) .* rand(rng, RealT, num_stages - 2)
```